src/diagrams/gantt/ganttDb.js

```diff
@@ -119,10 +141,9 @@ const getEndDate = function (prevTime, dateFormat, str) {
         d.add(durationStatement[1], 'weeks')
         break
     }
-    return d.toDate()
   }
   // Default date - now
-  return d.toDate()
+  return getNextValidDate(d, dateFormat, excludes)
```

Comment:
If I understand this correctly, this only checks whether an end date should be excluded, and if so, it returns the next Valid Date. But excluded dates can also fall in the duration of the task itself, and the duration should be adjusted accordingly. E.g. when you are excluding weekends, and the startdate is on a Friday, this function will return Tuesday as the enddate. But that's incorrect, it should return Thursday as the enddate.

Reply:
Yes. It was incorrect. My fix was to target the task after the start and end computation. See below.